### PR TITLE
PROD-2105 : Blog post search doesn't work correctly

### PR DESCRIPTION
**Pull request by milanhirapra**
### Jira Issue:

https://buddyboss.atlassian.net/browse/PROD-2105

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

File: src/bp-search/classes/class-bp-search-posts.php [hunk 1]
```diff
@@ -60,90 +83,91 @@ public function sql( $search_term, $only_totalrow_count = false ) {
...
+			// Get taxonomy.
 			$taxonomies = get_object_taxonomies( $this->pt_name );
 			foreach ( $taxonomies as $taxonomy ) {
 				if ( bp_is_search_post_type_taxonomy_enable( $taxonomy, $this->pt_name ) ) {
 					$tax[] = $taxonomy;
 				}
 			}
 
-			// Tax query left join
+			$select .= ' SELECT ';
```
Comment (jitendrabanjara1991) [inline src/bp-search/classes/class-bp-search-posts.php:102]: Remove space after SELECT

File: src/bp-search/classes/class-bp-search-posts.php [hunk 2]
```diff
@@ -60,90 +83,91 @@ public function sql( $search_term, $only_totalrow_count = false ) {
...
+
+			if ( $only_totalrow_count ) {
+				$select .= ' COUNT( DISTINCT id ) ';
+			} else {
+				$select .= $wpdb->prepare( ' DISTINCT id , %s as type, post_title LIKE %s AS relevance, post_date as entry_date ', $this->search_type, $placeholder );
```
Comment (jitendrabanjara1991) [inline src/bp-search/classes/class-bp-search-posts.php:107]: Remove space after DISTINCT id

**Review by jitendrabanjara1991 — CHANGES_REQUESTED** [review]

File: src/bp-search/classes/class-bp-search-posts.php (hunk 1 above)
Comment (milanhirapra) [reply src/bp-search/classes/class-bp-search-posts.php:102]: @jitendrabanjara1991 It's required when concat the `$select` variable

**Review by jitendrabanjara1991 — APPROVED** [review]
LGTM

**Pull request merged**